package/MDAnalysis/core/groups.py

```diff
@@ -1968,20 +1968,27 @@ def split(self, level):
 
         Parameters
         ----------
-        level : {'atom', 'residue', 'segment'}
+        level : {'atom', 'residue', 'molecule', 'segment'}
 
 
         .. versionadded:: 0.9.0
+        .. versionchanged:: 0.17.0
+           Added the 'molecule' level.
         """
         accessors = {'segment': 'segindices',
-                     'residue': 'resindices'}
+                     'residue': 'resindices',
+                     'molecule': 'molnums'}
 
         if level == "atom":
             return [self.universe.atoms[[a.ix]] for a in self]
 
         # higher level groupings
         try:
             levelindices = getattr(self, accessors[level])
+        except AttributeError:
+            raise ValueError('This universe does not have {} '
```

Comment:
I did hesitate. On one hand it is indeed an attribute missing that cause the problem; on the other hand, on the user point of view, it is the argument input that is the problem. I'll fix it.